package.json

```diff
@@ -20,17 +20,16 @@
   },
   "files": [
     "sample.resume.json",
-    "schema.json",
-    "validator.js"
+    "schema.json"
   ],
-  "dependencies": {
-    "z-schema": "^4.2.2"
-  },
   "devDependencies": {
+    "@babel/cli": "^7.8.4",
+    "@babel/core": "^7.9.0",
+    "@babel/preset-env": "^7.9.5",
     "ajv-cli": "^3.1.0",
-    "json-metaschema": "^1.3.0",
-    "tap-spec": "^5.0.0",
-    "tape": "^4.13.2"
+    "ajv": "^6.12.2",
+    "jest": "^25.4.0",
```

Comment:
Use Tape not Jest

Reply:
Okay, looks like moving to `jest` is the only remaining concern we have to address in this PR. Firstly, can you ack [my last comment on the subject](https://github.com/jsonresume/resume-schema/pull/371#issuecomment-618727111) before we proceed; I'd rather keep my points in single comments.

Some data on the concern of vulnerabilities, I just compared runs of [`audit`](https://classic.yarnpkg.com/en/docs/cli/audit/) on local clones of master of `jest` and `tape` and got this.


## jest:
```txt
➜  jest git:(master) yarn audit
...
5 vulnerabilities found - Packages audited: 919720
Severity: 3 Low | 1 Moderate | 1 High
```
| severity| type|
| -| -|
| low | Prototype Pollution |
| low           | Prototype Pollution |
| moderate      | Prototype Pollution |
| high          | Cross-Site Scripting |
| low           | Prototype Pollution |

## tape:
```txt
➜  tape git:(master) ✗ yarn audit
...
12 vulnerabilities found - Packages audited: 1956
Severity: 4 Low | 6 Moderate | 2 High
```

| severity| type|
| -| -|
| low           | Regular Expression Denial of Service                         |
| low           | Regular Expression Denial of Service                         |
| moderate      | Denial of Service                                            |
| high          | Code Injection                                               |
| low           | Prototype Pollution                                          |
| moderate      | Prototype Pollution                                          |
| moderate      | Prototype Pollution                                          |
| moderate      | Prototype Pollution                                          |
| moderate      | Prototype Pollution                                          |
| moderate      | Memory Exposure                                              |
| low           | Denial of Service                                            |
| high          | Insufficient Entropy                                         |

`tape` also requires that we use [`tap-spec`](https://www.npmjs.com/package/tap-spec). `audit` shows no vulns, but it was last published over two years ago.

`tape` was a reasonable choice when it was added to this project to run the first tests [over five years ago in 2014](https://github.com/jsonresume/resume-schema/commit/205074f). Javascript testing has evolved considerably since then, and it would be a mistake to ignore that. Just one example is [snapshot testing](https://jestjs.io/docs/en/snapshot-testing), and it would be immediately useful once we begin evolving the spec or adding migration functions. Jest also gives nice coverage reports with almost no extra configuration.

Reply:
I ack the previous comment. My take still stands.

I have used Jest plenty for FrontEnd dev, snapshot testing is great for comparing rendered DOM trees and such.

For super basic unit testing (ie output = result) Tape is a better fit. Testing tje schema or CLI really doesn't need to be more complex than that.

In terms of potential breaks (ie security or out-of-date deps), I was referring to surface area. Ie 919k vs 1.9k.

Tape outputs TAP test results, it's a standardized test reporting standard. Tap-spec is a TAP formatter. It just pretty prints the results. Feel free to remove it if you want.

If you want 'modern' use a lib that is Node + ESM compatible. Of which Jest isn't. But that isn't really necessary here unless the entire Node ecosystem deprecates CommonJS in the future.

I know Tape is old, I've been using it for OSS lib testing for years. It's also very stable, and lightweight for general-use unit testing. Which is exactly what we need here.

Keep in mind, nobody is paid to work on this project. Reducing the maintenance burden to a minimum is good for the long-term health and sustainability of the project.

Reply:
> In terms of potential breaks (ie security or out-of-date deps), I was referring to surface area. Ie 919k vs 1.9k.

How does my data from `audit` not trump this? Jest's vulnerabilities are more actively triaged and the software itself has a better maintenance record. It's also trusted by more projects in the javascript community than any other. ... and it's only devDependency, so this concern over vulns really doesn't matter at all since it doesn't get installed in dependent projects.

> I have used Jest plenty for FrontEnd dev, snapshot testing is great for comparing rendered DOM trees and such.

That something works great for FE jobs doesn't mean that it's _only_ for FE work. I've already cited babel as an example of significant non-FE project that uses jest for testing. Other examples of completely server-side and high-profile projects that use jest to test are: [webpack](https://github.com/webpack/webpack/blob/master/package.json#L117), [jest itself](https://github.com/facebook/jest/blob/master/package.json#L111), [`apollo-server`](https://github.com/apollographql/apollo-server/blob/master/package.json#L17), and [`ncc`](https://github.com/zeit/ncc/blob/master/package.json#L14). Of note, when I was browsing the most actively maintained projects for node, I didn't find any projects that still use `tape`.

> If you want 'modern' use a lib that is Node + ESM compatible. Of which Jest isn't.

Is tape? Jest respects `babel.config.js` which in practice is where the state of the art is. I have not come across any `.mjs` files in the real world since they were introduced several years ago.

I'm not after modern for modern's sake – if we're looking to get this project back on track and provide a nice experience for contributors, maybe we want to use something that most devs are likely to have used recently, and that also provides a first class experience for them?

> For super basic unit testing (ie output = result) Tape is a better fit. Testing the schema or CLI really doesn't need to be more complex than that.

Jest is as simple as we need it to be. Config defaults are sane and the expectation api is concise. And it can scale with the project as we need it to.

Also, I'd bet that the next time these test are re-written, we're going to want to have them generated (as was suggested in another thread.)

> Keep in mind, nobody is paid to work on this project.

I'd bet that most folks that find their way here from the professional world are probably more familiar with jest than they are with tape.

> Reducing the maintenance burden to a minimum is good for the long-term health and sustainability of the project.

Jest is not a maintenance burden.

My crystal ball says that we will eventually move to something more in line with the api that all the other testing frameworks have converged upon.

---

I'm done with this branch, but if you want to submit a PR against it that moves everything back to tape, that will be fine.

Reply:
> I have not come across any .mjs files in the real world since they were introduced several years ago.

I have [an entire org dedicated to ESM + Node](https://github.com/vanillaes). I'm also a member of the node/modules team. You don't see it in the wild b/c the new implementation was just unflagged in January and 'experimental' warnings were removed a week ago.

I've spent the past 2 years arguing about the 'future of JS'. TBH, I'm enjoying working on something that doesn't require arguing about the future of JS for once.

---

I understand you're not a fan. Don't worry about it. I'll take care of the rest.